provider/pdns/pdns.go

```diff
@@ -208,8 +207,7 @@ func (c *PDNSAPIClient) PatchZone(zoneID string, zoneStruct pgo.Zone) (resp *htt
 		return resp, err
 	}
 
-	log.Errorf("Unable to patch zone. %v", err)
-	return resp, err
+	return resp, provider.NewSoftError(fmt.Errorf("unable to patch zone: %v", err))
```

Comment:
When it fails when trying to update record, after multiple retries, I think this one should be a regular error.

Reply:
I think this should be a soft error. In a lot of cases, failing to update a record is due to user error. If the error is logged, then a user can "fix" the record definition and it can be applied on another run. Returning a regular error here bubbles up to a fatal error in the [controller](https://github.com/kubernetes-sigs/external-dns/blob/master/controller/controller.go#L338). Since the error is often recoverable, I think a soft error makes more sense. The AWS provider also returns a soft error when it cannot apply [some changes](https://github.com/kubernetes-sigs/external-dns/blob/master/provider/aws/aws.go#L698).